### PR TITLE
version up `dbt-labs/dbt_utils` as `1.0.0`

### PR DESCRIPTION
**Pull request by pei0804**
- https://github.com/elementary-data/elementary/issues/456

### CODE REVIEW

**Comment by pei0804** [comment]
if you no change behabior want to surrogate_key, You could use `surrogate_key_treat_nulls_as_empty_strings`.

**Comment by pei0804** [comment]
```console
❯ python ./run_unit_tests.py -f test_insensitive_get_dict_value -t snowflake
Running unit tests against target - snowflake
2022-12-06 16:33:55 — INFO — Running dbt --log-format json run-operation test_insensitive_get_dict_value --project-dir /Users/j-chikamori/go/src/github.com/pei0804/dbt-data-reliability/integration_tests --profiles-dir /Users/j-chikamori/.dbt --target snowflake

test_insensitive_get_dict_value:
1.SUCCESS
2.SUCCESS
3.SUCCESS
4.SUCCESS
5.SUCCESS
6.SUCCESS
7.SUCCESS
8.SUCCESS
9.SUCCESS
10.SUCCESS
11.SUCCESS
12.SUCCESS
13.SUCCESS
14.SUCCESS
```

**Comment by pei0804** [comment]
I used this PR code for the product code and it looks fine.
Is there anything else to do?

**Comment by IDoneShaveIt** [comment]
Hey @pei0804! Thank you for the contribution!
There are some changes that need to be done in order to have backwards compatibility (we want to make sure we support dbt versions 1.0.0 and above).
Reviewing the PR and adding the comments on the relevant places of the code 🙂

File: macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql [hunk 1]
```diff
@@ -111,7 +111,7 @@
         anomaly_scores as (
 
             select
-                {{ dbt_utils.surrogate_key([
+                {{ dbt_utils.generate_surrogate_key([
```
Comment (IDoneShaveIt) [inline macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql:114]: Older versions of dbt-utils don't have this macro, so we need to create a macro that is called generate_surrogate_key (similar to what we did [here](https://github.com/elementary-data/dbt-data-reliability/blob/master/macros/utils/cross_db_utils/current_timestamp.sql)).
Currently there is an issue with getting the installed version of dbt-utils so as a start lets do the following check:
Checks the dbt version, and if it is lower than 1.3.0 it calls `dbt_utils.surrogate_key` and if its equal or bigger it calls `dbt_utils.generate_surrogate_key`

(We will ask in our guide to set dbt utils package to be higher than 1.0.0 if working with dbt 1.3.0)

File: macros/utils/cross_db_utils/current_timestamp.sql [hunk 2]
```diff
@@ -2,9 +2,9 @@
   {% if dbt_version >= '1.2.0' %}
         {# This macro is depricated from dbt_utils version 0.9.0, but still hasn't got an equivalent macro at dbt-core #}
         {# Should be replaced to the equivalent macro once it released #}
-        {{ return(dbt_utils.current_timestamp()) }}
+        {{ return(dbt.current_timestamp()) }}
```
Comment (IDoneShaveIt) [inline macros/utils/cross_db_utils/current_timestamp.sql:5]: We need to use the backcompat macro to maintain the same behaviour as before 🙂 
```suggestion
        {{ return(dbt. current_timestamp_backcompat()) }}
```

File: macros/utils/cross_db_utils/current_timestamp.sql [hunk 3]
```diff
@@ -2,9 +2,9 @@
...
     {% else %}
-        {{ return(dbt_utils.current_timestamp()) }}
+        {{ return(dbt.current_timestamp()) }}
```
Comment (IDoneShaveIt) [inline macros/utils/cross_db_utils/current_timestamp.sql:7]: We need to use the backcompat macro to maintain the same behaviour as before 🙂
```suggestion
        {{ return(dbt. current_timestamp_backcompat()) }}
```

File: macros/utils/cross_db_utils/current_timestamp.sql [hunk 4]
```diff
@@ -2,9 +2,9 @@
   {% if dbt_version >= '1.2.0' %}
```
Comment (IDoneShaveIt) [inline macros/utils/cross_db_utils/current_timestamp.sql:2]: This method moved from dbt utils to dbt core on dbt core 1.3.0
```suggestion
  {% if dbt_version >= '1.3.0' %}
```

File: packages.yml [hunk 5]
```diff
@@ -1,3 +1,3 @@
 packages:
   - package: dbt-labs/dbt_utils
-    version: [">=0.8.0", "<1.0.0"]
+    version: [">=1.0.0", "<2.0.0"]
```
Comment (IDoneShaveIt) [inline packages.yml:3]: In order to support dbt 1.0.0 and above, we need to support older versions of dbt-utils as well:
```suggestion
    version: [">=0.8.0", "<2.0.0"]
```

**Review by IDoneShaveIt — CHANGES_REQUESTED** [review]
Thanks again for the contribution!
The PR looks great, I just added some comments to make the changes backwards compatibility 🙂

File: macros/utils/cross_db_utils/current_timestamp.sql (hunk 3 above)
Comment (pei0804) [reply macros/utils/cross_db_utils/current_timestamp.sql:7]: I see.

File: macros/utils/cross_db_utils/current_timestamp.sql (hunk 2 above)
Comment (pei0804) [reply macros/utils/cross_db_utils/current_timestamp.sql:5]: I trim whitespace.

File: macros/utils/cross_db_utils/generate_surrogate_key.sql [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{% macro generate_surrogate_key(field_list) -%}
```
Comment (pei0804) [inline macros/utils/cross_db_utils/generate_surrogate_key.sql:1]: I tried It. 
@IDoneShaveIt 
https://github.com/elementary-data/dbt-data-reliability/pull/174#discussion_r1040779731

File: macros/utils/cross_db_utils/current_timestamp.sql [hunk 7]
```diff
@@ -1,10 +1,10 @@
 {% macro current_timestamp() -%}
-  {% if dbt_version >= '1.2.0' %}
+  {% if dbt_version >= '1.3.0' %}
         {# This macro is depricated from dbt_utils version 0.9.0, but still hasn't got an equivalent macro at dbt-core #}
         {# Should be replaced to the equivalent macro once it released #}
-        {{ return(dbt_utils.current_timestamp()) }}
+        {{ return(dbt.current_timestamp_backcompat()) }}
     {% else %}
-        {{ return(dbt_utils.current_timestamp()) }}
+        {{ return(dbt.current_timestamp_backcompat()) }}
```
Comment (IDoneShaveIt) [inline macros/utils/cross_db_utils/current_timestamp.sql:7]: My bad, seems like I asked you to change it as well, but it should be:
```suggestion
        {{ return(dbt_utils.current_timestamp()) }}
```

File: macros/edr/data_monitoring/schema_changes/get_columns_changes_query.sql [hunk 8]
```diff
@@ -125,7 +125,7 @@
     )
 
         {# Creating a unique id for each row in the table, and adding execution id #}
-    select {{ dbt_utils.surrogate_key([
+    select {{ dbt_utils.generate_surrogate_key([
```
Comment (IDoneShaveIt) [inline macros/edr/data_monitoring/schema_changes/get_columns_changes_query.sql:128]: Missed it 🙂 
```suggestion
    select {{ generate_surrogate_key([
```

**Review by IDoneShaveIt — COMMENTED** [review]
Looking great!
2 small comments 🙂

**Comment by pei0804** [comment]
Thank you.
I fixed.
@IDoneShaveIt

**Comment by IDoneShaveIt** [comment]
Hey @pei0804!
Yeah everything is good 🙂 
We wanted to merge your PR and add our extension to it so no worries about the tests!

**Comment by pei0804** [comment]
👌

**Pull request merged**